File: src/main/java/com/github/alexandrecarlton/idea/settings/applier/impl/editor/spelling/SpellingSettingsApplier.java
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.spelling;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.layout.editor.spelling.SpellingSettings;
import com.intellij.spellchecker.settings.SpellCheckerSettings;

import java.nio.file.Path;

import javax.inject.Inject;

public class SpellingSettingsApplier implements SettingsApplier<SpellingSettings> {

  private final SpellCheckerSettings spellCheckerSettings;

  @Inject
  public SpellingSettingsApplier(SpellCheckerSettings spellCheckerSettings) {
    this.spellCheckerSettings = spellCheckerSettings;
  }

  @Override
  public void apply(SpellingSettings settings) {
    settings.dictionaries()
        .stream()
        .map(Path::toString)
        .forEach(spellCheckerSettings.getCustomDictionariesPaths()::add);
  }
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/applier/impl/editor/spelling/SpellingSettingsApplier.kt
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.spelling

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier
import com.github.alexandrecarlton.idea.settings.layout.editor.spelling.SpellingSettings
import com.intellij.spellchecker.SpellCheckerManager
import com.intellij.spellchecker.settings.SpellCheckerSettings
import javax.inject.Inject

class SpellingSettingsApplier @Inject
constructor(
    private val spellCheckerManager: SpellCheckerManager,
    private val spellCheckerSettings: SpellCheckerSettings
) : SettingsApplier<SpellingSettings> {

    override fun apply(settings: SpellingSettings) {
        settings.dictionaries
            ?.map { it.absolutePath }
            ?.let { spellCheckerSettings.customDictionariesPaths = it }
        settings.acceptedWords
            ?.let(spellCheckerManager::updateUserDictionary)
    }
}
